l3build.lua

```diff
@@ -131,6 +131,10 @@ if forcedocepoch then
     typesetopts = typesetopts .. " -utc"
   end
 end
+-- Allow for LaTeX "dev" release
+if options["dev"] and checkformat == "latex" then
+  checkformat = "latex-dev"
```

Comment:
not sure if it helps or if it makes things worse, but shouldn't there perhaps a terminal output stating --dev not applied if checkformat is not latex?

Reply:
I've added one - it only shows at the top level due to the way configs work, but hopefully that's enough?